kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/metrics/ScanReportSuite.scala

```diff
@@ -0,0 +1,481 @@
+      case (Some(found), Some(expected)) =>
+        assert(found.getName == expected.getName && found.getChildren == expected.getChildren)
+      case (found, expected) => assert(found == expected)
+    }
+
+    // Since we cannot know the actual duration of the scan we sanity check that they are > 0 and
+    // less than the total operation duration
+    assert(scanReport.getScanMetrics.getTotalPlanningDurationNs > 0)
+    assert(scanReport.getScanMetrics.getTotalPlanningDurationNs < durationNs)
+
+    assert(scanReport.getScanMetrics.getNumAddFilesSeen == expectedNumAddFiles)
+    assert(scanReport.getScanMetrics.getNumAddFilesSeenFromDeltaFiles ==
+      expectedNumAddFilesFromDeltaFiles)
+    assert(scanReport.getScanMetrics.getNumActiveAddFiles == expectedNumActiveAddFiles)
+    assert(scanReport.getScanMetrics.getNumDuplicateAddFiles == expectedNumDuplicateAddFiles)
+    assert(scanReport.getScanMetrics.getNumRemoveFilesSeenFromDeltaFiles ==
+      expectedNumRemoveFilesSeenFromDeltaFiles)
+  }
+
+  test("ScanReport: basic case with no extra parameters") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // Set up delta table with 1 add file
+      spark.range(10).write.format("delta").mode("append").save(path)
+
+      checkScanReport(
+        path,
+        expectException = false,
+        expectedNumAddFiles = 1,
+        expectedNumAddFilesFromDeltaFiles = 1,
+        expectedNumActiveAddFiles = 1
+      )
+    }
+  }
+
+  test("ScanReport: basic case with read schema") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // Set up delta table with 1 add file
+      spark.range(10).withColumn("c2", col("id") % 2)
+        .write.format("delta").mode("append").save(path)
+
+      checkScanReport(
+        path,
+        expectException = false,
+        expectedNumAddFiles = 1,
+        expectedNumAddFilesFromDeltaFiles = 1,
+        expectedNumActiveAddFiles = 1,
+        readSchema = Some(new StructType().add("id", LongType.LONG))
+      )
+    }
+  }
+
+  test("ScanReport: different filter scenarios") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // Set up partitioned table
+      spark.range(10).withColumn("part", col("id") % 2)
+        .write.format("delta").partitionBy("part").save(path)
+
+      val partFilter = new Predicate("=", new Column("part"), Literal.ofLong(0))
+      val dataFilter = new Predicate("<=", new Column("id"), Literal.ofLong(0))
+      val expectedSkippingFilter = new Predicate(
+        "<=", new Column(Array("minValues", "id")), Literal.ofLong(0))
+
+      // No filter - 2 add files one for each partition
+      checkScanReport(
+        path,
+        expectException = false,
+        expectedNumAddFiles = 2,
+        expectedNumAddFilesFromDeltaFiles = 2,
+        expectedNumActiveAddFiles = 2
+      )
+
+      // With partition filter
```

Comment:
are we expecting any of these filters to impact any of the scan metrics? e.g. expectedNumAddFiles expectedNumAddFilesFromDeltaFiles expectedNumActiveAddFiles ?

Reply:
no we only increment the file counts before filtering; we can add additional metrics for filtering in the future (but this is where it would be really helpful for something like a SV to have an implicit count field or something)

Reply:
Okay! It would be nice if you made it easier for the reader to know that we do _not_ expect scans with partition/data filters to have different scan metrics than scans without the filters.

You could do this via a comment or even save `noFilterScanReport = .....` (idk if that method returns a scan report, but tit could) and then do `checkScanReport(..., expectedNumAddFiles = noFilterScanReport.numAddFiles)` etc.

You choose which you think is best

Reply:
Will LGTM after this!